antarest/study/storage/rawstudy/raw_study_service.py

```diff
@@ -402,11 +402,38 @@ def unarchive(self, study: RawStudy) -> None:
         Raises:
             BadArchiveContent: If the archive is corrupted or in an unknown format.
         """
-        with open(self.get_archive_path(study), mode="rb") as fh:
+        with open(self.find_archive_path(study), mode="rb") as fh:
             self.import_study(study, fh)
 
-    def get_archive_path(self, study: RawStudy) -> Path:
-        return Path(self.config.storage.archive_dir / f"{study.id}.zip")
+    def find_archive_path(self, study: RawStudy) -> Path:
+        """
+        Fetch for archive path of a study if it exists else raise an incorrectly archived study.
+
+        Args:
+            study: The study to get the archive path for.
+
+        Returns:
+            The full path of the archive file (zip or 7z).
+        """
+        archive_dir: Path = self.config.storage.archive_dir
+        for suffix in [".zip", ".7z"]:
+            path = archive_dir.joinpath(f"{study.id}{suffix}")
+            if path.is_file():
+                return path
+        raise FileNotFoundError(f"Study {study.id} archiving process is corrupted (no archive file found).")
+
+    def infer_archive_path(self, study: RawStudy) -> Path:
+        """
+        Create archive path of a study.
+
+        Args:
+            study: The study to get the archive path for.
+
+        Returns:
+            The full path of the archive file (7z).
+        """
+        archive_dir: Path = self.config.storage.archive_dir
+        return archive_dir.joinpath(f"{study.id}.7z")
```

Comment:
This hard coded ".7z" is too far away from the implementation where we actually create the a 7z archive.
This will be difficult to maintain. If we want to change again the archiving format, we will need to find again all those places.